packages/medusa/src/repositories/discount-condition.ts

```diff
@@ -108,39 +117,218 @@ export class DiscountConditionRepository extends Repository<DiscountCondition> {
+      .where("discon.discount_rule_id = :discountRuleId", {
+        discountRuleId,
+      })
+      .getMany()
+
+    // in case of no discount conditions, we assume that the discount
+    // is valid for all
+    if (!discountConditions.length) {
+      return true
+    }
+
+    // retrieve all conditions for each type where condition type id is in jointable (products, product_types, product_collections, product_tags)
+    // "E.g. for a given product condition, give me all products affected by it"
+    // for each of these types, we check:
+    //    if condition operation is `in` and the query for conditions defined for the given type is empty, the discount is invalid
+    //    if condition operation is `not_in` and the query for conditions defined for the given type is not empty, the discount is invalid
+    for (const condition of discountConditions) {
+      const { joinTableForeignKey } = this.getJoinTableResourceIdentifiers(
+        condition.type
+      )
+
+      // Tags is a ManyToMany relation, so we need to check all tags that exist
+      // For each tag, we perform the operation outlined in above comment
+      if (
+        joinTableForeignKey ===
+        DiscountConditionJoinTableForeignKey.PRODUCT_TAG_ID
+      ) {
+        for (const tag of product.tags) {
+          const tagConds = await this.queryConditionTable({
+            type: condition.type,
+            condId: condition.id,
+            resourceId: tag.id,
+          })
+
+          if (
+            condition.operator === DiscountConditionOperator.IN &&
+            !tagConds.length
+          ) {
+            return false
+          }
+
+          if (
+            condition.operator === DiscountConditionOperator.NOT_IN &&
+            tagConds.length
+          ) {
+            return false
+          }
+        }
+      } else {
+        let resourceId: string | null = null
+
+        if (condition.type === DiscountConditionType.PRODUCTS) {
+          resourceId = product.id
+        }
+
+        if (condition.type === DiscountConditionType.PRODUCT_TYPES) {
+          resourceId = product.type_id
+        }
+
+        if (condition.type === DiscountConditionType.PRODUCT_COLLECTIONS) {
+          resourceId = product.collection_id
+        }
+
+        if (!resourceId) {
+          return false
+        }
+
+        const resourceSpecificConditions = await this.queryConditionTable({
+          type: condition.type,
+          condId: condition.id,
+          resourceId,
+        })
+
+        if (
+          condition.operator === DiscountConditionOperator.IN &&
+          !resourceSpecificConditions.length
+        ) {
+          return false
+        }
+
+        if (
+          condition.operator === DiscountConditionOperator.NOT_IN &&
+          resourceSpecificConditions.length
+        ) {
+          return false
+        }
+      }
+    }
+
+    return true
+  }
+
+  async canApplyForCustomer(
+    discountRuleId: string,
+    customer: Customer
```

Comment:
similar to the comment on #1172 - I'd suggest that we make this work with just a `customer_id`. The query would look almost identical to the product tag query.